GVRf/Framework/jni/objects/scene_object.cpp

```diff
@@ -199,4 +200,27 @@ void SceneObject::set_visible(bool visibility = true) {
     }
 }
 
+bool SceneObject::isColliding(SceneObject *scene_object) {
+
+    //Get the transformed bounding boxes in world coordinates and check if they intersect
+    //Transformation is done by the getTransformedBoundingBoxInfo method in the Mesh class
+
+    float thisobj_bbox[6], checkobj_bbox[6];
+
+    glm::mat4 thisobj_model_matrix = this->render_data()->owner_object()->transform()->getModelMatrix();
+    memcpy(thisobj_bbox, (this->render_data()->mesh()->getTransformedBoundingBoxInfo(&thisobj_model_matrix)), sizeof(float)*6);
+
+    glm::mat4 checkobj_model_matrix = scene_object->render_data()->owner_object()->transform()->getModelMatrix();
+    memcpy(checkobj_bbox, (scene_object->render_data()->mesh()->getTransformedBoundingBoxInfo(&checkobj_model_matrix)), sizeof(float)*6);
+
```

Comment:
Same concern here. I don't see any reason why you need to do memcpy here since you are just doing comparison in the following.